src/Bicep.Core.IntegrationTests/TemplateEvaluator.cs

```diff
@@ -20,6 +20,72 @@ namespace Bicep.Core.IntegrationTests
 {
     public class TemplateEvaluator
     {
+        private class TemplateEvaluationContext : IEvaluationContext
```

Comment:
The class in this PR is very similar to the one in `Bicep.Cli/Services`. We should consolidate these classes into one and move it to Bicep.Core or another shared project. I don't plan to address this in the current PR. Created an issue to track it as a tech debt: https://github.com/Azure/bicep/issues/14772.